src/components/header/SearchInput.tsx

```diff
@@ -16,22 +17,41 @@ import {
   searchInputField,
   searchInputSelect,
 } from '@styles/header/headerStyles';
-import { useCallback, useState } from 'react';
+import { ChangeEvent, useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { SelectArrowIcon } from './SelectArrowIcon';
 
 export default function SearchInput() {
-  const [selectState, setSelectState] = useState('All Categories');
+  const [selectedCategory, setSelectedCategory] = useState('All Categories');
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const dispatch = useDispatch();
+
   const { data: categories } = useGetCategoriesQuery();
   const selectChangeHandler = useCallback((event: SelectChangeEvent) => {
-    setSelectState(event.target.value);
+    setSelectedCategory(event.target.value);
   }, []);
 
+  useEffect(() => {
+    const timer = setTimeout(() => {
```

Comment:
just interesting, why do you need a timer over here?

Reply:
to dispatch filter action after the user stopped typing
at first, I dispatched on every change

Reply:
as I said, debounce )) You can read about throttle and debounce. You will use them often. https://codepen.io/goit-academy/pen/rNyryZP

Reply:
okay)
thank you